File: LeetCode/CodingInterviews/Java/GetKthFromEnd.java
package LeetCode;

class Solution {
    public ListNode getKthFromEnd(ListNode head, int k) {
        ListNode res;
        ListNode p = head;
        for (int i = 0; i < k; i++)
            p = p.next;
        res = head;
        while (p != null) {
            p = p.next;
            res = res.next;
        }
        return res;
    }
}

File: LeetCode/CodingInterviews/Go/getKthFromEnd.go
package coding

func getKthFromEnd(head *ListNode, k int) *ListNode {
	var p, res = head, head
	for k > 0 {
		if p == nil {
			return nil
		}
		p = p.Next
		k--
	}
	for p != nil {
		p = p.Next
		res = res.Next
	}
	return res
}
